File: java/Array/11_Container_With_Most_Water.java
/**
    time: O(n)
    two pointers
 */

public static int maxArea(int[] height) {
    int left = 0, right = height.length - 1;
    int maxS = 0;
    while (left < right) {
        int h, tmp = (right - left);
        if (height[left] < height[right]) {
            h = height[left];
            left++;
        } else {
            h = height[right];
            right--;
        }
        tmp = tmp * h;
        if (maxS < tmp) maxS = tmp;
    }
    return maxS;
}

File: python3/11_Container_With_Most_Water.py
from typing import List
import pdb


class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        area = 0
        while left < right:
            if height[left] >= height[right]:
                area = max(area, (right - left) * height[right])
                right -= 1
            else:
                area = max(area, (right - left) * height[left])
                left += 1
        return area
            

if __name__ == "__main__":
    lst = [1, 1]
    print(Solution().maxArea(lst))
